### PR TITLE
Lastupdated filter

### LINKED ISSUE #90: Update the LocationHierarchy endpoint to filter locations by `_lastUpdated` when on `list` mode
## Context
- The LocationHierarchy hierarchy endpoint currently returns all the locations every time it is called. While this is good when trying to fetch all the data, it's kinda unnecessary when using it for subsequent sync on the app.  
- We would like to introduce the `_lastUpdated` parameter to solve the above problem.  

## Implementation 
- [ ] This parameter would accept a DateTime and then pass the same to HAPI using the `_lastUpdated` parameter on HAPI.  
- [ ] The parameter should only be relevant if the LocationHierarchy request also has `mode=list`
- [ ] In case the `_lastUpdated` is not set or is empty the endpoint should behave as it does right now which is to fetch everything

## Acceptance Criteria
- [ ] Users should be able to use the `_lastUpdated` parameter after this work is completed
- [ ] In case the `_lastUpdated` is not added the endpoint should work as is right now i.e it should return everything
- [ ] The `_lastUpdated` parameter should only be relevant is the location hierarchy request has `mode=list`

### PR DESCRIPTION
**Pull request by lincmba**
Add lastupdated filter to the locationhierarchy endpoint on list mode
Refactor code

Resolves #90 

**Engineer Checklist**

- [x] I have written **Unit tests** for any new feature(s) and edge cases for
      bug fixes
- [x] I have added documentation for any new feature(s) and configuration
      option(s) on the `README.md`
- [x] I have run `mvn spotless:check` to check my code follows the project's
      style guide
- [x] I have run `mvn clean test jacoco:report` to confirm the coverage report
      was generated at `plugins/target/site/jacoco/index.html`
- [x] I ran `mvn clean package` right before creating this pull request.

### CODE REVIEW

File: plugins/src/main/java/org/smartregister/fhir/gateway/plugins/LocationHierarchyEndpointHelper.java [hunk 1]
```diff
@@ -56,13 +59,18 @@ public LocationHierarchy getLocationHierarchy(
...
         if (CacheHelper.INSTANCE.skipCache()) {
             locationHierarchy =
                     getLocationHierarchyCore(
-                            locationId, preFetchAdminLevels, postFetchAdminLevels, filterInventory);
+                            locationId,
+                            preFetchAdminLevels,
+                            postFetchAdminLevels,
+                            filterInventory,
+                            lastUpdated);
         } else {
```
Comment (ndegwamartin) [inline plugins/src/main/java/org/smartregister/fhir/gateway/plugins/LocationHierarchyEndpointHelper.java:74]: Lets remove caching here for correctness then we can enhance with a new caching approach on a separate PR cc @dubdabasoduba

**Comment by lincmba** [comment]
closing in favor of https://github.com/onaio/fhir-gateway-extension/pull/97
